src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -612,6 +613,12 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
+        /// <summary>
+        /// Indicates whether the connection should be closed if the server throws an exception.
+        /// </summary>
+        /// <returns>A <see cref="bool"/> indicating if the streams should be closed.</returns>
+        protected virtual bool ShouldCloseStreamOnException() => this.closeStreamOnException;
```

Comment:
I'm thinking the implementation might want to make its decision based on the `Exception` object. Can we pass that in as a parameter?